SplitFiles/face.c

```diff
@@ -0,0 +1,182 @@
+#include <arch/zx.h>
+#include <stdio.h>
+#include "util.h"
+#include "face.h"
+
+void happyFace(const int x, const int y)
+{
+    int px = (x - 1) * 8;
+    int py = (y - 1) * 8;
+    *zx_pxy2saddr(px + 8, py) = 0x0;
```

Comment:
Having established the address top line of an 8x8 character block on the screen using pxy2saddr it _must_ be quicker to use simple addition to find the address of the next 7 tows of that block - ie add 8x32 = 256 the the pointer?  Surely.

Reply:
Also would it not be more readable/maintainable to have a single function to write these things on screen and just pass a pointer to the pixel data in to that?  Not sure the optimizer will realise its the same code over and over and our tap file is getting big now...28k is "big" for a Spectrum!

Reply:
yes.

Reply:
Thing to be aware of, though, is that you are, roughly speaking, executing three times as much code with a simple line line loop as with the unrolled version - so we might want to watch performance